src/compile.ts

```diff
@@ -55,38 +58,103 @@ export function compile(
       );
 
       function visitor(node: ts.Node): ts.Node {
-        try {
-          if (isAppsyncResolver(node)) {
-            return visitAppsyncResolver(node);
-          } else if (isReflectFunction(node)) {
-            return toFunction("FunctionDecl", node.arguments[0]);
+        const _visit = () => {
+          try {
+            if (isAppsyncResolver(node)) {
+              return visitAppsyncResolver(node);
+            } else if (isReflectFunction(node)) {
+              return toFunction("FunctionDecl", node.arguments[0]);
+            } else if (isEventBusWhenFunction(node)) {
+              return visitEventBusWhen(node);
+            } else if (isEventBusMapFunction(node)) {
+              return visitEventBusMap(node);
+            }
+            return node;
+          } catch (err) {
+            const error =
+              err instanceof Error ? err : Error("Unknown compiler error.");
+            return newExpr("Err", [
+              ts.factory.createNewExpression(
+                ts.factory.createIdentifier(error.name),
+                undefined,
+                [ts.factory.createStringLiteral(error.message)]
+              ),
+            ]) as unknown as Statement;
           }
-        } catch (err) {
-          const error =
-            err instanceof Error ? err : Error("Unknown compiler error.");
-          return newExpr("Err", [
-            ts.factory.createNewExpression(
-              ts.factory.createIdentifier(error.name),
-              undefined,
-              [ts.factory.createStringLiteral(error.message)]
-            ),
-          ]) as unknown as Statement;
-        }
-        return ts.visitEachChild(node, visitor, ctx);
+        };
+        // keep processing the children of the updated node.
+        return ts.visitEachChild(_visit(), visitor, ctx);
```

Comment:
Why visit the transformed code? This would be expensive and unnecessary.

Reply:
because the child of the transformed class is not transformed and may need to be.

```ts
bus
   .when(event => event)
   .map(event => event.detail)
```

The current visit is on the `map` method. The contents of map are updated and returned, but the fact that it was chained off a `when` was not considered during `visitEventBusMapFunction`. 

We could add the exploration down the children to each of the visit methods, but that seems unnecessary.

This is something you wouldn't hit when always transforming the top of the statement (`new AppResolver()`), but once you need to transform a method on a class its needed.